### PR TITLE
[VOL-112] Running point symlink

### PR DESCRIPTION
**Pull request by robhaswell**
https://clusterhq.atlassian.net/browse/VOL-112

@brendan-donegan was the primary author of this branch, I cleared up a recursive loop in it which allowed all the tests to pass.

### CODE REVIEW

File: pkg/containers/docker.go [hunk 1]
```diff
@@ -17,36 +20,80 @@ func isRelated(volume string, container *docker.Container) bool {
...
+		delete(runtime.stopped[volume], cid)
+	}
 	return nil
 }
 
-func (runtime DockerRuntime) Stop(volume string) error {
+func (runtime *DockerRuntime) attemptStop(cid ContainerID) error {
+	var err error
+	for idx := 0; idx < 10; idx++ {
+		// TODO: Need the ID here so refactoring neccesary
```
Comment (lukemarsden) [inline pkg/containers/docker.go]: I don't understand this comment. Is it actionable?

File: pkg/containers/docker.go [hunk 2]
```diff
@@ -17,36 +20,80 @@ func isRelated(volume string, container *docker.Container) bool {
...
+		if err = runtime.client.StopContainer(string(cid), 10); err == nil {
```
Comment (lukemarsden) [inline pkg/containers/docker.go]: move magic number into const?

File: pkg/api/api_test.go [hunk 3]
```diff
@@ -44,3 +44,18 @@ func TestSwitchVolume(t *testing.T) {
...
+		t.Errorf("Could not create TempDir: %s\n", err)
+	}
+	dvol := NewDvolAPI(DvolAPIOptions{
+		BasePath:                 basePath,
+		DisableDockerIntegration: true,
+	})
+	if err := dvol.CheckoutBranch("foo", "master", "bar", true); err != nil {
+		t.Errorf("CheckoutBranch failed: %s\n", err)
+	}
+	// TODO: Get the active branch
```
Comment (lukemarsden) [inline pkg/api/api_test.go]: why?

File: pkg/containers/docker.go (hunk 1 above)
Comment (robhaswell) [inline pkg/containers/docker.go]: I don't understand either. This looks like standard "retry 10 times" code (questionable if 10 is the right number however).

I will add the logging statements around this.

**Comment by lukemarsden** [comment]
Did you need to add the retry logic to make some tests pass?

Looks generally fine, please improve the comments, remove enough logging to make https://travis-ci.org/ClusterHQ/dvol/jobs/120087932 pass and then merge if travis is happy. Thanks!

File: pkg/containers/docker.go (hunk 2 above)
Comment (robhaswell) [inline pkg/containers/docker.go]: Great minds - I did this before reading your comment while addressing some surrounding issues :)

**Comment by robhaswell** [comment]
> Did you need to add the retry logic to make some tests pass?

I don't know, but I suspect not.

> please improve the comments

I did.

> remove enough logging

Also done.

**Comment by lukemarsden** [comment]
Looks like travis has caught a couple of minor issues holding this up:
* https://travis-ci.org/ClusterHQ/dvol/jobs/120094081#L5021 - one of the golang tests fails now
* https://travis-ci.org/ClusterHQ/dvol/jobs/120094082#L429 - go vet is no longer happy

**Pull request merged**